### PR TITLE
[CALCITE] Add support for CLOSE cursor call in CREATE PROCEDURE statement 

### PR DESCRIPTION
**Pull request by michaelvdow**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/SqlCloseCursor.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Parse tree for {@code SqlCloseCursor} call.
+ */
+public class SqlCloseCursor extends SqlCall {
+  private static final SqlSpecialOperator OPERATOR =
+      new SqlSpecialOperator("CLOSE", SqlKind.CLOSE);
```
Comment (dasch-1) [inline core/src/main/java/org/apache/calcite/sql/SqlCloseCursor.java:30]: Should this be something like SqlKind.CLOSE_CURSOR (same for the text string)?

File: core/src/main/java/org/apache/calcite/sql/SqlCloseCursor.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+   *
+   * @param pos SQL parser position
+   * @param cursorName Name of the cursor to close, must not be null
+   */
+  public SqlCloseCursor(SqlParserPos pos, SqlIdentifier cursorName) {
+    super(pos);
+    this.cursorName = Objects.requireNonNull(cursorName);
+  }
+
+  @Override public void unparse(SqlWriter writer, int leftPrec, int rightPrec) {
```
Comment (dasch-1) [inline core/src/main/java/org/apache/calcite/sql/SqlCloseCursor.java:45]: nit: Line wrap.

File: core/src/main/java/org/apache/calcite/sql/SqlCloseCursor.java (hunk 1 above)
Comment (michaelvdow) [reply core/src/main/java/org/apache/calcite/sql/SqlCloseCursor.java:30]: Yeah I think that would be clearer, will update

File: core/src/main/java/org/apache/calcite/sql/SqlCloseCursor.java (hunk 2 above)
Comment (michaelvdow) [reply core/src/main/java/org/apache/calcite/sql/SqlCloseCursor.java:45]: In my editor it looks like this line is just at 80 chars.

Comment (dasch-1) [reply core/src/main/java/org/apache/calcite/sql/SqlCloseCursor.java:45]: You're right - looks good.

**Pull request merged**